spec/linter-eslint-spec.js

```diff
@@ -128,21 +140,70 @@ describe('The eslint provider for Linter', () => {
     })
   })
 
-  describe('Fix errors when saved', () => {
+  describe('Fix errors', () => {
+    let editor
+    let doneCheckingFixes
+    let tempFixtureDir
+    let tempFixturePath
+    let tempConfigPath
+
     beforeEach(() => {
-      atom.config.set('linter-eslint.fixOnSave', true)
+      waitsForPromise(() => {
+        tempFixtureDir = fs.mkdtempSync(tmpdir() + path.sep)
+        tempFixturePath = `${tempFixtureDir}${path.sep}fixed.js`
+        tempConfigPath = `${tempFixtureDir}${path.sep}.eslintrc.yaml`
+        fs.createReadStream(configPath).pipe(fs.createWriteStream(tempConfigPath))
+
+        return atom.workspace.open(fixPath).then((openEditor) => {
+          openEditor.saveAs(tempFixturePath)
+          editor = openEditor
+        })
+      })
     })
-    it('should fix lint errors when saved', () => {
-      waitsForPromise(() =>
-        atom.workspace.open(fixPath).then((editor) => {
-          lint(editor).then((messages) => {
-            expect(messages.length).toBe(2)
-            editor.save()
-            lint(editor).then((messagesAfterSave) => {
-              expect(messagesAfterSave.length).toBe(0)
-            })
-          })
+
+    afterEach(() => {
+      rimraf.sync(tempFixtureDir)
+    })
+
+    it('should fix linting errors', () => {
+      function firstLint(textEditor) {
+        return lint(textEditor)
+        .then((messages) => {
+          // The original file has two errors
+          expect(messages.length).toBe(2)
+          return textEditor
         })
+      }
+      function makeFixes(textEditor) {
+        return fix(textEditor)
+          .then((messagesAfterSave) => {
+            // Linter reports a successful fix
+            expect(messagesAfterSave).toBe('Linter-ESLint: Fix Complete')
+            return textEditor
```

Comment:
Nothing is being chained after this, so this isn't necessary. It's not harming anything though so it's fine to leave in.